spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/filters/SimpleRouteLocator.java

```diff
@@ -51,8 +55,10 @@
 	private String dispatcherServletPath = "/";
 	private String zuulServletPath;
 
-	private AtomicReference<Map<String, ZuulRoute>> routes = new AtomicReference<>();
+	private AtomicReference<Map<String, List<ZuulRoute>>> routes = new AtomicReference<>();
 	private int order = DEFAULT_ORDER;
+	
+	private UrlPathHelper urlPathHelper = new UrlPathHelper();
```

Comment:
This came from `PreDecorationFilter` and `LocationRewriteFilter`. Both were using `urlPathHelper.getPathWithinApplication(ctx.getRequest())` to get the requestUri.

The following piece of code is important as well to get the correct requestUri. `this.urlPathHelper.setRemoveSemicolonContent(properties.isRemoveSemicolonContent());`

Since extraction of requestUri has now moved to SimpleRouteLocator, this was required to move as well.